wpilibj/src/athena/java/edu/wpi/first/wpilibj/QuantityEncoder.java

```diff
@@ -0,0 +1,80 @@
+package edu.wpi.first.wpilibj;
+
+import tec.uom.se.quantity.Quantities;
+
+import javax.measure.Quantity;
+import javax.measure.Unit;
+
+/**
+ *
+ * @param <Q> The Quantity being measured. (ex: Length, Angle)
+ * @param <S> The Speed of the Quantity being measured. (ex: Speed, AngularSpeed)
+ */
+// Package private so it is not part of the public API
+abstract class QuantityEncoder<Q extends Quantity<Q>, S extends Quantity<S>> {
+
+  private final Encoder m_encoder;
```

Comment:
I think it is going to be a better user experience to deprecate Encoder and write methods that take either a Distance or Angle encoder.  If we were to extend Encoder, users could change values like distancePerPulse and then units stop working.  In the end, I would like to change QuantityEncoder to not have an encoder but rather call the appropriate HAL methods itself.